### PR TITLE
Add Hackage in purl-spec

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: TG1999 <tushar.goel.dav@gmail.com>

### CODE REVIEW

**Comment by TG1999** [comment]
Oops, I did some mistake :p, will do it fine by tommorow

**Comment by TG1999** [comment]
Done @stevespringett, now PR is good to go, please have a look :D

File: README.rst [hunk 1]
```diff
@@ -616,6 +616,18 @@ candidate list further down.
         pkg:golang/github.com/gorilla/context@234fd47e07d1004f0aed9c#api
 
 
+- `hackage` for Haskell packages
+
+  - The default repository is `https://hackage.haskell.org`.
+  - The `version` is package version.
+  - The `name` is case sensitive and use kebab-case
```
Comment (pombredanne) [inline README.rst:623]: Is there a doc that tells something about case sensitivity? Also can you explain  what is  kebab-case?

Comment (TG1999) [reply README.rst:623]: Yeah sure, like snake case use '_' kebab case use '-', more info can be found here

Comment (TG1999) [reply README.rst:623]: https://medium.com/better-programming/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841

**Comment by TG1999** [comment]
Yes, I will like to add tests, but once want to have my implementation reviewed, please can you review this first https://github.com/package-url/packageurl-python/pull/26 :D

File: README.rst (hunk 1 above)
Comment (TG1999) [reply README.rst:623]: I don't have any doc, but I tried to find a package name by changing it's letter cases, it does not identified it, 'AC-HalfInteger' & 'ac-halfInteger' were not same, so I concluded it from here

**Comment by stevespringett** [comment]
Looks good to me and the test cases pass.

**Comment by TG1999** [comment]
Cool @stevespringett

**Comment by pombredanne** [comment]
@TG1999 sorry for having waited so long to merge this.. But do you mind to rebase, and move this to the new types document instead?

**Review by pombredanne — APPROVED** [review]
LGTM!
Thanks

**Pull request merged**